tests/core/tests/test_resources.py

```diff
@@ -2570,3 +2571,143 @@ def test_has_natural_foreign_key(self):
                 resources.has_natural_foreign_key(model),
                 expected_result
             )
+
+
+class ManyToManyWidgetDiffTest(TestCase):
+    fixtures = ["category", "book"]
+
+    def setUp(self):
+        pass
+
+    def test_m2m_on_new_entity(self):
+        # issue 1558
```

Comment:
This comment would be better at the class level, with a one line description of what we are testing:

> issue 1558 - test cases when m2m fields are null